webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -79,15 +106,28 @@ def update_correlation_plot(
                 continue
             valid_responses = []
             for response in responses:
-                response_df = ensemble.responses[response].data_df()
-                if response_df.empty:
+                response_el = ensemble.responses[response]
+                if response_el.empty:
                     continue
-                x_index = corr_xindex.get(response, 0)
-                parameter_df[response] = response_df.iloc[x_index]
-                valid_responses.append(response)
-
+                if response_el.observations:
+                    for obs in response_el.observations:
+                        for idx, obs_idx in enumerate(selected_obs.get(obs.name, [])):
+                            if obs.axis_type == AxisType.TIMESTAMP:
+                                resp_index = str(Timestamp(obs_idx))
+                            else:
+                                resp_index = obs_idx
+                            parameter_df[
+                                f"{response_el.name}::{idx}"
+                            ] = response_el.data.loc[resp_index]
+                            valid_responses.append(f"{response_el.name}::{idx}")
+                else:
+                    # TODO: consider removing this else clause not really
+                    #  any point is showing responses without observations but here for now
+                    parameter_df[response] = response_el.data.iloc[0]
+                    valid_responses.append(response)
             corrdf = parameter_df.corr(method=correlation_metric)
             corrdf = corrdf.drop(valid_responses, axis=0).fillna(0)
+            selected_response = f"{selected_response}::0"
```

Comment:
I don't understand this. Can you maybe comment why we need this "transformation"?

Reply:
This is a hack that will need to be fixed. Currently, it is needed because now that `valid_responses` are of the form `f"{response.name}::{idx}"` so it is impossible to check if the selected_response is part of the valid responses without this transformation.

I am sure this part of the code could be improved.